app/domain/util/rate_limited_cache.rb

```diff
@@ -62,13 +63,26 @@ def recalculate(args)
         )
         return
       end
-      @cache[args] = @target.call(**args)
+      @cache[cache_key] = @target.call(**args)
       @logger.debug(LogMessages::Util::RateLimitedCacheUpdated.new)
-      @refresh_history[args].push(@time.now)
+      @refresh_history[cache_key].push(@time.now)
+    end
+
+    def cached_key(args)
+      if args.has_key?(:cache_key)
+        cache_key = args.fetch(:cache_key)
+        args.delete(:cache_key)
+        return cache_key
+      end
+      args
     end
 
     def too_many_requests?(args)
+      @logger.debug("tamir @refresh_history before prune")
```

Comment:
delete those messages and consider to add a real log message

Reply:
deleted